src/coreclr/tests/src/JIT/opt/InstructionCombining/CondToBranchlessExpr.cs

```diff
@@ -0,0 +1,284 @@
+using System;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+public class Program
+{
+    public static int returnCode = 100;
+
+    public static int Main(string[] args)
+    {
+        var testsWithOpt = new Tests();
+        var testsWithoutOpt = new TestsWithoutOptimization();
+
+        var allTestsWithOpt = testsWithOpt.GetType().GetMethods()
+            .Where(m => m.Name.StartsWith("Test"))
+            .OrderBy(m => m.Name)
+            .ToArray();
+
+        var allTestsWithoutOpt = testsWithoutOpt.GetType().GetMethods()
+            .Where(m => m.Name.StartsWith("Test"))
+            .OrderBy(m => m.Name)
+            .ToArray();
+
+        long[] testInput = Enumerable.Range(-256, 256)
+            .Select(i => (long)i)
+            .Concat(new [] { (long)int.MaxValue - 1, int.MaxValue, long.MaxValue - 1, long.MaxValue })
+            .ToArray();
+
+        for (int i = 0; i < allTestsWithOpt.Length; i++)
+        {
+            for (int j = 0; j < testInput.Length; j++)
+            {
+                object[] invokeArgs;
+                Type argType = allTestsWithOpt[i].GetParameters()[0].ParameterType;
+                if (argType == typeof(int))
+                    invokeArgs = new object[] {unchecked((int) testInput[j])};
```

Comment:
>invokeArgs = new object[] {unchecked((int) testInput[j])}; [](start = 20, length = 58)

I am not sure if i completely understand this. Could you please elaborate why this is needed?

Reply:
I see what you are doing. Please ignore my previous comment.

---
In reply to: [388538116](https://github.com/dotnet/runtime/pull/32716#discussion_r388538116) [](ancestors = 388538116)